node/node.go

```diff
@@ -742,6 +751,16 @@ func (node *AlgorandFullNode) GetPendingTxnsFromPool() ([]transactions.SignedTxn
 	return bookkeeping.SignedTxnGroupsFlatten(node.transactionPool.PendingTxGroups()), nil
 }
 
+// ensureParticipationDB opens or creates a participation DB.
+func ensureParticipationDB(genesisDir string) (account.ParticipationRegistry, error) {
+	accessorFile := filepath.Join(genesisDir, config.ParticipationRegistryFilename)
+	accessor, err := db.MakeAccessor(accessorFile, false, false)
```

Comment:
avoid reinventing the wheel - use `db.Pair` which would get you reader/writer accessors.